src/main/java/com/meilisearch/sdk/Documents.java

```diff
@@ -21,82 +22,162 @@ protected Documents(Config config) {
     /**
      * Retrieves the document at the specified index uid with the specified identifier
      *
+     * @param <T> Type of the document returned
      * @param uid Partial index identifier for the requested documents
      * @param identifier ID of the document
-     * @return String containing the requested document
+     * @param targetClass Class of the document returned
+     * @return Object containing the requested document
+     * @throws MeilisearchException if the client request causes an error
+     */
+    <T> T getDocument(String uid, String identifier, Class<T> targetClass)
+            throws MeilisearchException {
+        URLBuilder urlb = new URLBuilder();
+        urlb.addSubroute("indexes")
+                .addSubroute(uid)
+                .addSubroute("documents")
+                .addSubroute(identifier);
+        String urlPath = urlb.getURL();
+        return httpClient.<T>get(urlPath, targetClass);
+    }
+
+    /**
+     * Retrieves the document at the specified index uid with the specified identifier
+     *
+     * @param uid Partial index identifier for the requested documents
+     * @param identifier ID of the document
+     * @param param accept by the documents route
+     * @param targetClass Class of the document returned
+     * @return Object containing the requested document
      * @throws MeilisearchException if client request causes an error
      */
-    String getDocument(String uid, String identifier) throws MeilisearchException {
-        String urlPath = "/indexes/" + uid + "/documents/" + identifier;
-        return httpClient.get(urlPath, String.class);
+    <T> T getDocument(String uid, String identifier, DocumentsQuery param, Class<T> targetClass)
+            throws MeilisearchException {
+        URLBuilder urlb = new URLBuilder();
+        urlb.addSubroute("indexes")
+                .addSubroute(uid)
+                .addSubroute("documents")
+                .addSubroute(identifier)
+                .addParameter("limit", param.getLimit())
+                .addParameter("offset", param.getOffset())
+                .addParameter("fields", param.getFields());
```

Comment:
This piece of code will repeat in all the occurrences where `DocumentsQuery` will be used. So why not create a method `toQuery` or something inside of the `DocumentsQuery`, then you can just call it.

`return httpClient.<T>get(param.toQuery(), targetClass);`